clap-utils/src/input_parsers.rs

```diff
@@ -96,15 +96,18 @@ pub fn pubkeys_sigs_of(matches: &ArgMatches<'_>, name: &str) -> Option<Vec<(Pubk
 }
 
 // Return a signer from matches at `name`
+#[allow(clippy::type_complexity)]
 pub fn signer_of(
-    name: &str,
     matches: &ArgMatches<'_>,
+    name: &str,
     wallet_manager: Option<&Arc<RemoteWalletManager>>,
-) -> Result<Option<Box<dyn Signer>>, Box<dyn std::error::Error>> {
+) -> Result<(Option<Box<dyn Signer>>, Option<Pubkey>), Box<dyn std::error::Error>> {
     if let Some(location) = matches.value_of(name) {
-        signer_from_path(matches, location, name, wallet_manager).map(Some)
+        let signer = signer_from_path(matches, location, name, wallet_manager)?;
+        let signer_pubkey = signer.pubkey();
+        Ok((Some(signer), Some(signer_pubkey)))
```

Comment:
`Ok(Some((signer, pubkey))`, then next chunk is unneeded ?

Reply:
I tried the return type `Result<Option<(Box<dyn Signer>, Pubkey)>, err>` first, but it was a pain on the caller side, as the option needs destructuring for signer and pubkey to be used separately in `generate_unique_keypairs()` and `index_of()` respectively.

Reply:
Hmm, yeah that is annoying :confused: 

Nit rescinded